http/oidc/src/main/java/org/wildfly/security/http/oidc/ServerRequest.java

```diff
@@ -241,4 +245,33 @@ public String getError() {
         }
     }
 
+    public static AccessAndIDTokenResponse getBearerToken(OidcClientConfiguration oidcClientConfiguration, String username, String password) throws Exception {
+        AccessAndIDTokenResponse tokenResponse;
+        HttpClient client = oidcClientConfiguration.getClient();
```

Comment:
I haven't checked object access in all the new methods but should we be having some Assert.checkNotNull calls for the mandatory parameters?

Reply:
This is called from private API and will be called with valid parameters. In general though, it would probably be good to do a pass through these types of methods to add checkNotNull calls where appropriate. I'll add some good first issues for that I think.